### ISSUE TITLE
Document how to register new packages

### ISSUE THREAD

**Issue by fridex** [thoth-station/thoth-application#2485]
**Is your feature request related to a problem? Please describe.**

As @codificat pointed out, let's have documentation that states how to register Python packages to the system. This documentation could state how to register Python packages and also what is happening in the system when a new package is solved.

**Comment by goern** [comment]
:+1: 

as this is a user-facing doc, and we would like to have one sig in the lead
/remove-sig stack-guidance

**Comment by codificat** [comment]
> as this is a user-facing doc, and we would like to have one sig in the lead

I agree that one SIG should lead. However, given the content of the documentation (expected to heavily rely on knowledge of how solver/advise works) I would say this should be driven by
/sig stack-guidance
rather than
/remove-sig user-experience

Also, one thing I would like to add: beyond the purely technical details, I believe the documentation should mention the requirements that packages are expected to meet: what do we accept, what not, and why.
